### ISSUE TITLE
Cannot connect to remote ssh host after upgrade (whitespace in Windows username)

### ISSUE THREAD

**Issue by fbartels** [microsoft/vscode-remote-release#2547]
- VSCode Version: 1.43.0
- OS Version: Windows 10 10.0.17763

Steps to Reproduce:

1. Upgrade to 1.43.0
2. Have windows home folder with a space
3. connect to a system that requires ssh keys and is stored in your `.ssh/config`

The following is logged:

```
[13:18:29.946] Log Level: 2
[13:18:29.956] remote-ssh@0.50.0
[13:18:29.956] win32 x64
[13:18:29.959] SSH Resolver called for "ssh-remote+remote-server", attempt 1
[13:18:29.960] SSH Resolver called for host: remote-server
[13:18:29.960] Setting up SSH remote "remote-server"
[13:18:29.964] Acquiring local install lock: C:\Users\FELIXB~1\AppData\Local\Temp\vscode-remote-ssh-remote-server-install.lock
[13:18:29.992] Looking for existing server data file at c:\Users\Felix Bartels\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-ssh\vscode-ssh-host-remote-server-78a4c91400152c0f27ba4d363eb56d2835f9903a\data.json
[13:18:29.995] Using commit id "78a4c91400152c0f27ba4d363eb56d2835f9903a" and quality "stable" for server
[13:18:29.996] Install and start server if needed
[13:18:30.007] Checking ssh with "ssh -V"
[13:18:30.083] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[13:18:30.084] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[13:18:30.145] > OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5
[13:18:30.145] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[13:18:30.378] > OpenSSH_7.9p1, OpenSSL 1.0.2p  14 Aug 2018
[13:18:30.381] askpass server listening on \\.\pipe\vscode-ssh-askpass-1e53feba2800e952d897f64b316a81fe81f3a214-sock
[13:18:30.382] Spawning local server with {"ipcHandlePath":"\\\\.\\pipe\\vscode-ssh-askpass-7a075b36caf5392d3164e31714c2cbc42b439bbe-sock","sshCommand":"C:\\Program Files\\Git\\usr\\bin\\ssh.exe","sshArgs":["-v","-T","-D","54126","remote-server"],"dataFilePath":"c:\\Users\\Felix Bartels\\AppData\\Roaming\\Code\\User\\globalStorage\\ms-vscode-remote.remote-ssh\\vscode-ssh-host-remote-server-78a4c91400152c0f27ba4d363eb56d2835f9903a\\data.json"}
[13:18:30.382] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"c:\\Users\\Felix Bartels\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.50.0\\out\\local-server\\askpass.bat","VSCODE_SSH_ASKPASS_NODE":"C:\\Users\\Felix Bartels\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe","VSCODE_SSH_ASKPASS_MAIN":"c:\\Users\\Felix Bartels\\.vscode\\extensions\\ms-vscode-remote.remote-ssh-0.50.0\\out\\askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"\\\\.\\pipe\\vscode-ssh-askpass-1e53feba2800e952d897f64b316a81fe81f3a214-sock"}
[13:18:30.462] Spawned 25928
[13:18:30.593] > local-server> Spawned ssh: 24868
[13:18:30.638] stderr> OpenSSH_7.9p1, OpenSSL 1.0.2p  14 Aug 2018
[13:18:30.870] stderr> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:jhqlVHf8oBUOkQiMT5zBevNGtATZtVAc7KJ9L5dbwvA
[13:18:31.014] stderr> 'C:\Users\Felix' is not recognized as an internal or external command,
[13:18:31.015] stderr> operable program or batch file.
[13:18:31.049] stderr> 'C:\Users\Felix' is not recognized as an internal or external command,
[13:18:31.049] stderr> operable program or batch file.
[13:18:31.079] stderr> Permission denied, please try again.
[13:18:31.111] stderr> 'C:\Users\Felix' is not recognized as an internal or external command,
[13:18:31.112] stderr> operable program or batch file.
[13:18:31.144] stderr> Permission denied, please try again.
[13:18:31.178] stderr> 'C:\Users\Felix' is not recognized as an internal or external command,
[13:18:31.178] stderr> operable program or batch file.
[13:18:31.209] stderr> fbartels@remote-server.domain.com: Permission denied (publickey,password).
[13:18:31.224] > local-server> ssh child died, shutting down
[13:18:31.244] Local server exit: 0
[13:18:31.245] Received install output: OpenSSH_7.9p1, OpenSSL 1.0.2p  14 Aug 2018
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:jhqlVHf8oBUOkQiMT5zBevNGtATZtVAc7KJ9L5dbwvA
'C:\Users\Felix' is not recognized as an internal or external command,
operable program or batch file.
'C:\Users\Felix' is not recognized as an internal or external command,
operable program or batch file.
Permission denied, please try again.
'C:\Users\Felix' is not recognized as an internal or external command,
operable program or batch file.
Permission denied, please try again.
'C:\Users\Felix' is not recognized as an internal or external command,
operable program or batch file.
fbartels@remote-server.host-consultants.de: Permission denied (publickey,password).

[13:18:31.246] Resolver error: Permission denied (publickey,password).
[13:18:31.250] ------
```
`remote-server` is a system that is defined in my `~/.ssh/config`

Does this issue occur when all extensions are disabled?: Yes/No

As this is a problem with an extension I did not try the above.

**Comment by mwalden2004** [comment]
I am able to reproduce this issue on Windows 10, after the most recent update.

**Comment by MitchTalmadge** [comment]
Can confirm that this issue exists in my username "Mitch Talmadge" on windows: 

```
'C:\Users\Mitch' is not recognized as an internal or external command,
operable program or batch file.
Host key verification failed.
```

**Comment by tangwudu** [comment]
reinstall "Remote-SSH" to version 0.48.0 solve my problem

**Comment by Uguudei** [comment]
My vscode stopped connecting to remote server and showed same error suddenly today. 
Reinstalling "Remote-SSH" to  version 0.49.0 solved my problem. Latest version 0.50.0 caused this problem

**Comment by Lewis2z** [comment]
I have been having issues with the upgrade and further to this, it looks like it is attempting to use the Windows username as the user in the ssh connection.

I looked at the auth.log file on my VM and found this:
```
Mar 19 18:01:10 dashboard sshd[28994]: Invalid user Lewis from 192.168.100.1 port 60612
Mar 19 18:01:10 dashboard sshd[28994]: Connection closed by invalid user Lewis 192.168.100.1 port 60612 [preauth]
```
In the extension output (in VSCode) for v50 it was attempting to connect as `Lewis@192.168.100.100` instead of `vagrant@192.168.100.100` which the v49 of this extension was doing.

Downgrading to version 49 solved all my problems!

**Comment by roblourens** [comment]
I believe this was fixed in https://github.com/microsoft/vscode-remote-release/issues/2406. Someone let me know if there is still an issue.
